--- datasource.py-orig	2021-05-30 00:34:50.230000000 +0000
+++ datasource.py	2021-05-30 00:32:41.840000000 +0000
@@ -166,7 +166,7 @@
     return util.check_call_make(path, targets=['clip'], variables=variables_items)
 
 
-def seed(cache_dir=CACHE_DIR, product=DEFAULT_PRODUCT, bounds=None, max_download_tiles=9, **kwargs):
+def seed(cache_dir=CACHE_DIR, product=DEFAULT_PRODUCT, bounds=None, max_download_tiles=900, **kwargs):
     """Seed the DEM to given bounds.
 
     :param cache_dir: Root of the DEM cache folder.
